Src/ILGPU/IR/Construction/CompareOperations.tt

```diff
@@ -69,6 +69,8 @@ namespace ILGPU.IR.Construction
             var isUnsigned = (flags & CompareFlags.UnsignedOrUnordered) == CompareFlags.UnsignedOrUnordered;
             switch (left.BasicValueType)
             {
+                case BasicValueType.Float16:
+                    return CreatePrimitiveValue(location, <#= operation.Prefix #>left.Float16Value<#= operation.Operation #>right.Float32Value<#= operation.Suffix #>);
```

Comment:
```suggestion
                    return CreatePrimitiveValue(location, <#= operation.Prefix #>left.Float16Value<#= operation.Operation #>right.Float16Value<#= operation.Suffix #>);
```

Reply:
Nice one. I accidentally merged the wrong lines during rebasing.